LLama/Experimental/Common/SequenceOutput.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LLama.Experimental.Common
+{
+    /// <summary>
+    /// The model output associated with a sequence.
+    /// </summary>
+    public class SequenceOutput
+        // TODO: Beam search
+    {
+        /// <summary>
+        /// The output token ID.
+        /// </summary>
+        public int OutputTokenId { get; init; }
+
+        /// <summary>
+        /// The ID of the parent sequence (for forking in beam search).
+        /// </summary>
+        public int ParentSeqId { get; init; }
+
+        /// <summary>
+        /// The logprobs of the output token.
+        /// (Token id -> logP(x_i+1 | x_0, ..., x_i))
+        /// </summary>
+        public float[]? Logprobs { get; init; }
```

Comment:
If this is used for _every_ output token it would mean a huge number of allocations (an array of VocabSize floats for every token)!

Reply:
Yes, I'm considering something like [this](https://github.com/SciSharp/LLamaSharp/pull/683/files#diff-721ca68c35eb0a60867eab939dcdd247390cef060fd06a6b343fd8099274d801R11), only get it from native when it's needed.